sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -322,13 +325,15 @@ static void _on_sock_udp_evt(sock_udp_t *sock, sock_async_flags_t type, void *ar
             memcpy(&_listen_buf[cursor], stackbuf, res);
             cursor += res;
         }
+
+        /* make sure we reply with the same address that the request was destined for */
         gcoap_socket_t socket = { .type = GCOAP_SOCKET_TYPE_UDP, .socket.udp = sock };
-        _process_coap_pdu(&socket, &remote, _listen_buf, cursor, truncated);
+        _process_coap_pdu(&socket, &remote, (void*)&aux, _listen_buf, cursor, truncated);
```

Comment:
You cannot rely on `sock_udp_aux_rx_t` and `sock_udp_aux_tx_t` being compatible. E.g. when module `sock_aux_rssi` is in use, the member `flags` will have a different offset.

You could reorder the struct members so that this cast would work, but please add a `static_assert()` using [`offsetof()`](https://en.cppreference.com/w/c/types/offsetof) to ensure that the `flags` and the `local` member do indeed have the same offset in the struct. But IMO it is better to just copy the address to a temporary `sock_udp_aux_tx_t` instead.

Also, `sock_udp_recv_buf_aux()` will clear flags, if it was able to supply the corresponding auxiliary value. So `SOCK_AUX_GET_LOCAL` needs to be re-set.